editor/editor_node.cpp

```diff
@@ -2356,6 +2356,10 @@ void EditorNode::_edit_current(bool p_skip_foreign) {
 			} else {
 				if (FileAccess::exists(base_path + ".import")) {
 					editable_info = TTR("This resource belongs to a scene that was imported, so it's not editable.\nPlease read the documentation relevant to importing scenes to better understand this workflow.");
+				} else {
+					if ((!get_edited_scene() || get_edited_scene()->get_scene_file_path() != base_path) && ResourceLoader::get_resource_type(base_path) == "PackedScene") {
```

Comment:
That's the same `else` branch as in the `if (!base_path.is_resource_file()) {` block. You could invert the logic to simplify:
```cpp
String base_path = current_res->get_path().substr(0, subr_idx);
if (FileAccess::exists(base_path + ".import")) {
	if (!base_path.is_resource_file()) {
		if (get_edited_scene() && get_edited_scene()->get_scene_file_path() == base_path) {
			info_is_warning = true;
		}
		editable_info = TTR("This resource belongs to a scene that was imported, so it's not editable.\nPlease read the documentation relevant to importing scenes to better understand this workflow.");
	} else {
		editable_info = TTR("This resource belongs to a scene that was imported, so it's not editable.\nPlease read the documentation relevant to importing scenes to better understand this workflow.");
	}
} else if ((!get_edited_scene() || get_edited_scene()->get_scene_file_path() != base_path) && ResourceLoader::get_resource_type(base_path) == "PackedScene") {
	editable_info = TTR("This resource belongs to a scene that was instantiated or inherited.\nChanges to it must be made inside the original scene.");
}
```
(to be double checked)